### PR TITLE
self-checkout: use dedicated endpoints

### LINKED ISSUE #927: self-checkout: use dedicated endpoints
Use dedicated endpoints for creating a loan via self-checkout. Add UI for displaying self-checkout delivery method for the loans on the backoffice.

### PR DESCRIPTION
**Pull request by ntarocco**
* define new ad-hoc search and checkout endpoints, to be able to have
  better control on contraints and input and output payloads
* use delivery methods to store when the checkout is a self-checkout
* closes https://github.com/CERNDocumentServer/cds-ils/issues/927

### CODE REVIEW

File: invenio_app_ils/circulation/templates/invenio_app_ils_circulation/notifications/self_checkout.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{% block title %}
```
Comment (ntarocco) [inline invenio_app_ils/circulation/templates/invenio_app_ils_circulation/notifications/self_checkout.html:1]: To be added to CDS-ILS too

File: invenio_app_ils/circulation/config.py [hunk 2]
```diff
@@ -162,7 +162,13 @@
             dest="ITEM_ON_LOAN",
             trigger="checkout",
             transition=ILSToItemOnLoan,
-            permission_factory=loan_checkout_permission,
+            permission_factory=backoffice_permission,
```
Comment (ntarocco) [inline invenio_app_ils/circulation/config.py:165]: Reverted (https://github.com/inveniosoftware/invenio-app-ils/pull/1213/files#diff-7466ae9bfb0eff430d949f02d2b515743ddd72c6c0ee5552a8019fcb7f80f1e6R165)

File: invenio_app_ils/errors.py [hunk 3]
```diff
@@ -194,9 +205,47 @@ def __init__(self, patron_pid, current_user_pid, **kwargs):
...
+class LoanSelfCheckoutItemInvalidStatus(LoanSelfCheckoutItemUnavailable):
+    """A patron cannot self-checkout an item that cannot circulate."""
+
+    supportCode = "SELF-CHECKOUT-001"
+
+
+class LoanSelfCheckoutDocumentOverbooked(LoanSelfCheckoutItemUnavailable):
+    """A patron cannot self-checkout an item for an overbooked document."""
+
+    supportCode = "SELF-CHECKOUT-002"
```
Comment (kpsherva) [inline invenio_app_ils/errors.py:236]: I think we should share these codes with the librarians, since they will be the first point of contact

**Review by kpsherva — APPROVED** [review]
Thank you for this massive work!

File: invenio_app_ils/errors.py (hunk 3 above)
Comment (ntarocco) [reply invenio_app_ils/errors.py:236]: Yes, that was the idea indeed ;)

**Review by sakshamarora1 — APPROVED** [review]
LGTM! 🚀

File: invenio_app_ils/circulation/views.py [hunk 4]
```diff
@@ -157,6 +199,48 @@ def post(self, **kwargs):
...
+        try:
+            barcode = request.args["barcode"].upper()
+            assert barcode
+        except (KeyError, AssertionError):
+            msg = "Parameter `barcode` is missing"
+            raise MissingRequiredParameterError(description=msg)
+
+        try:
+            pid, item = self_checkout_get_item_by_barcode(barcode)
+        except ItemCannotCirculateError:
```
Comment (ntarocco) [inline invenio_app_ils/circulation/views.py:221]: The msg `Item not found ` is not very clear, as the user does not know what is an Item.
We should catch `ItemNotFound` here, and re-raise an exception with a different exc `LoanSelfCheckoutItemNotFound`, e.g. `Literature not found. Please try again with another barcode or contact the library.`

Comment (anikachurilova) [reply invenio_app_ils/circulation/views.py:221]: Done ✅  and updated screeshots here: https://github.com/inveniosoftware/react-invenio-app-ils/pull/649

**Pull request merged**